packages/useModal/src/useModalHook.tsx

```diff
@@ -43,30 +35,40 @@ const isFunctionalComponent = (Component: React.FunctionComponent) => {
 /**
  * React hook for showing modal windows
  */
-export const useModalHook = (component: ModalType, inputs: any[] = []): [ShowModal, HideModal] => {
+export const useModalHook = (
+  component: React.FunctionComponent<any>,
+  deps?: React.DependencyList
+): [ShowModal, HideModal] => {
   if (!isFunctionalComponent(component)) {
     throw new Error(
       'Only stateless components can be used as an argument to useModal. You have probably passed a class component where a function was expected.'
     )
   }
 
-  const key = useMemo(generateModalKey, [])
-  const modal = useMemo(() => component, inputs)
-  const context = useContext(ModalContext)
-  const [isShown, setShown] = useState<boolean>(false)
-  const showModal = useCallback(() => setShown(true), [])
-  const hideModal = useCallback(() => setShown(false), [])
+  const MemoizedComponent = useMemo(() => component, deps)
+  const rootRef = useRef<HTMLDivElement | null>()
 
-  useEffect(() => {
-    if (isShown) {
-      context.showModal(key, modal)
-    } else {
-      context.hideModal(key)
+  const showModal = useCallback(() => {
+    const el = document.createElement('div')
+    el.setAttribute('data-id', 'harness-modal-root')
```

Comment:
if there are multiple modals open, data-id will not be unique.. should we add some unique key here?

Reply:
done